api/envoy/api/v2/route/route.proto

```diff
@@ -959,6 +972,14 @@ message RetryPolicy {
 
   // HTTP headers which must be present in the request for retries to be attempted.
   repeated HeaderMatcher retriable_request_headers = 10;
+
+  // Specifies a limit on concurrent retries in relation to the number of active requests. This
+  // parameter is optional.
+  //
+  // .. note::
+  //
+  //    If this field is set, the retry budget will override any configured retry circuit breaker.
+  RetryBudget retry_budget = 11;
```

Comment:
Q: Right now we can set a retry policy on either the virtual host or the route, with the route overriding the virtual host. I don't feel strongly about this, but I'm wondering if it would be a better experience to allow the retry budget to be set independently from the policy. In this case, if the user sets a per-route policy, they will have to duplicate the budget across every policy because I don't think anything is merged from the virtual host. WDYT? @snowp?

Reply:
It would be nice to have a cluster-level retry budget config and have each route optionally override it. It might get confusing to reason about where retry overflows are coming from, since we're using the retry_overflow counter for both the max_retries circuit breaker and the retry budget.

I can start adding this top-level budget to a different branch and defer to you as to whether it should go in this patch or a subsequent one along with runtime knobs and the solution to the HTTP/1.1 request tracking problem below.

Reply:
Do you think this is going to be problematic at Lyft? Or you think we can just configure this universally? I'm concerned that it's going to cause problems with us for our own usage right away?

Reply:
I'm pretty confident this can just be configured universally.

Reply:
With the config you have now how can we configure this universally?